### PR TITLE
Integrated channel api logs loading fix

### PR DESCRIPTION
**Pull request by MueezKhan246**
Description:
Adding get_queryset for fix of integrated channel api logs loading.

JIRA:

### CODE REVIEW

File: integrated_channels/integrated_channel/admin/__init__.py [hunk 1]
```diff
@@ -130,5 +130,9 @@ class IntegratedChannelAPIRequestLogAdmin(admin.ModelAdmin):
 
     list_per_page = 20
 
+    def get_queryset(self, request):
+        queryset = super().get_queryset(request)
+        return queryset.select_related('enterprise_customer')
```
[the remark this answers is not in the document]
Comment (MueezKhan246) [reply integrated_channels/integrated_channel/admin/__init__.py:135]: @sameen by default django does not allow the combination of values or values_list with select_related

[the remark this answers is not in the document]
Comment (MueezKhan246) [reply integrated_channels/integrated_channel/admin/__init__.py:135]: https://internal.courses.stage.edx.org/admin/integrated_channel/integratedchannelapirequestlogs/
the list filter is displaying to much data on loading 
i believe at least `time_taken` can be removed from it

[the remark this answers is not in the document]
Comment (hamzawaleed01) [reply integrated_channels/integrated_channel/admin/__init__.py:135]: Just discussed with Mueez - I think we can simply just remove whole `list_filter`. It doesn't seem to be useful in this model. We already have `search_fields` in place.

https://github.com/openedx/edx-enterprise/blob/a4bb35e5de9b8d05724407d587384cf4154dc727/integrated_channels/integrated_channel/admin/__init__.py#L106

[the remark this answers is not in the document]
Comment (sameenfatima78) [reply integrated_channels/integrated_channel/admin/__init__.py:129]: Sounds good!

File: integrated_channels/integrated_channel/models.py [hunk 2]
```diff
@@ -907,14 +920,21 @@ class IntegratedChannelAPIRequestLogs(TimeStampedModel):
...
     )
 
     class Meta:
         app_label = "integrated_channel"
         verbose_name_plural = "Integrated channels API request logs"
+        indexes = [
+            models.Index(
+                fields=['channel_name', 'status_code', 'enterprise_customer_configuration_id', 'endpoint',],
+            )
+        ]
```
Comment (sameenfatima78) [inline integrated_channels/integrated_channel/models.py:937]: Curious to know why did we choose these fields for indexing?

Comment (sameenfatima78) [reply integrated_channels/integrated_channel/models.py:937]: @MueezKhan246 I think we can try to test the changes with the optimized query first. If that doesn't resolve the issue then we can look into adding these indexes. I'm just trying to be cautious just so we don't add any redundant indexes.

Comment (zamanafzal) [reply integrated_channels/integrated_channel/models.py:937]: Agree, let's test the changes with the optimized query first.

Comment (MueezKhan246) [reply integrated_channels/integrated_channel/models.py]: @sameenfatima78 @zamanafzal i have removed the index for now

**Review by sameenfatima78 — APPROVED** [review]

File: integrated_channels/integrated_channel/models.py (hunk 2 above)
Comment (sameenfatima78) [reply integrated_channels/integrated_channel/models.py]: Just approved your PR

**Review by zamanafzal — APPROVED** [review]

**Pull request merged**